cms/static/js/models/group_configuration.js

```diff
@@ -87,22 +88,29 @@ function(Backbone, _, str, gettext, GroupModel, GroupCollection) {
                 };
             }
 
-            if (attrs.groups.length < 1) {
+            if (!this.canBeEmpty && attrs.groups.length < 1) {
                 return {
                     message: gettext('There must be at least one group.'),
                     attributes: { groups: true }
                 };
             } else {
                 // validate all groups
-                var invalidGroups = [];
+                var validGroups = [],
+                    invalidGroups = [];
                 attrs.groups.each(function(group) {
-                    if(!group.isValid()) {
+                    if (!group.isValid()) {
                         invalidGroups.push(group);
+                    } else {
+                        if (_.find(validGroups, function(validGroup) { return validGroup.get('name') === group.get('name'); })) {
+                            invalidGroups.push(group);
+                        } else {
+                            validGroups.push(group);
+                        }
                     }
                 });
                 if (!_.isEmpty(invalidGroups)) {
                     return {
-                        message: gettext('All groups must have a name.'),
+                        message: gettext('All groups must have a unique name.'),
```

Comment:
@catong @cahrens could you verify this error message? I intended it to be used when the user either a) didn't give a group a name or b) used the same name twice.